### PR TITLE
[AJ-1352] Add reference aliases to allow for custom naming with species prefix.

### PR DESCRIPTION
**Pull request by rickymagner**
This PR replaces #4222, instead merging from an internal branch rather than my external fork.

## Summary of changes:
(See discussion in comments below on evolution of ideas for this PR. The end result is still the same, but the implementation has been changed to now avoid making specific reference aliases, and instead using one method to generate reference *labels* from hardcoded reference metadata.)

This PR makes a few changes to how reference data names get displayed in Terra. In particular, it introduces the notion of aliases for references to disentangle the backend label for a reference bundle from what the user sees in the front end on the Data tab. These changes should be fully backwards compatible, still maintaining the old convention of accessing the data in workflows via `workspace.referenceData_{old_backend_name}`. 

### What
The key differences introduced are:
* A `reference_aliases.ts` file which provides dictionaries mapping the alias name to the backend label, and vice versa.
* Add aliases for all existing references to fit the format `species: ref` so users can see at a glance which species are supported. 
* Update dropdown for selecting reference data to use the aliases.
* Update text displayed after user has imported reference data to show the alias rather than the backend label.
* Update the tooltip on the "delete" button for reference data to show the alias name.

### Why
This allows greater flexibility in maintaining references on Terra. Now you can easily update/change the user-side names of references as appropriate. For example, if references continue to accrue on Terra, one might want to change the UI to cluster references by organism (somewhat similar to how this PR suggests using the new `species: ref` paradigm). This PR sets up the appropriate infrastructure to allow for this kind of option without worrying about breaking backwards compatibility in user workflows.

### Testing strategy
I tested it works locally using `yarn start`. :)

### Visual Aids
image

### CODE REVIEW

File: src/data/reference-data/reference-aliases.ts
Comment (nawatts) [inline src/data/reference-data/reference-aliases.ts]: A style convention: Names starting with upper case letters are usually reserved for classes / components. Suggest lower casing this name.

```suggestion
export const refAliasToName = {
```

Also, since other code tends to use the full "reference" vs abbreviating, I'd suggest doing the same here.

```suggestion
export const referenceAliasToName = {
```

**Review by nawatts — CHANGES_REQUESTED** [review]
My understanding is that we can't hide the actual name of the reference used in the attribute keys because that's what users will enter in workflow configuration to use the reference data.

As a result of this, all the aliases added here follow the format of aliasing a reference named X to "<species> <X>". They add additional information to the reference name but also still include the reference name.

Because of that limitation, I'm not sure we want to introduce the concept of a general alias. I think this approach could make it too easy to add an alias that omits the actual name.

Instead, could we add additional metadata for references? Something like:
```ts
const referenceSpecies = {
  hg38: 'Human',
  'Mmul-10': 'Monkey',
  ...
```

Or more generally:
```ts
const referenceMetadata = {
  hg38: {
    species: 'Human',
    ...
  },
  'Mmul-10': {
    species: 'Monkey',
    ...
  },
  ...
```

The labels shown for each reference could render that extra information. And having the reference metadata in a more structured format (name and species vs everything in the name) would make it easier to do things like sort by species. It would also mean that this PR only has to touch the code for rendering the labels, not for modifying the workspace attributes, making this a lower risk change.

What do you think?

**Comment by rickymagner** [comment]
@nawatts Thanks for your helpful suggestions. I just pushed some changes that hopefully address them all. These include:

- More strict format for rendering reference aliases. In particular, the method `makeReferenceAlias` now controls exactly how to format aliases to the user in the UI menus, and they all follow the formula `<species>: <label>`. 
- The `reference_aliases.ts` is now `reference-metadata.ts` which follows the format you suggested. I like that it's flexible enough to potentially add other metadata in the future which could interact with the UI in interesting ways, like e.g. versioning, short descriptions, etc. 
- I switched to using camelCase generally. The one exception I kept was importing `ReferenceData` from `references.ts`, since this is done elsewhere.

Otherwise everything else should function the same as before. I also did a quick switch of the order in `references.ts` to put hg38 above hg19/37, in hopes the subliminal messaging will encourage people to stop using the now very old human reference...

File: src/data/reference-data/ReferenceDataImporter.js
Comment (nawatts) [inline src/data/reference-data/ReferenceDataImporter.js]: This is the type of thing I'd like to avoid. We should never have to go backwards from a reference label to a reference ID. All state should be based on the reference ID and then there can be a one-way mapping from ID to label.

File: src/data/reference-data/ReferenceDataImporter.js
Comment (nawatts) [inline src/data/reference-data/ReferenceDataImporter.js]: To accomplish that, instead of passing only reference labels as options to Select, let's separate the `value` and `label` for each option.

```ts
options: _.keys(ReferenceData).map((referenceName) => ({ value: referenceName, label: ... }),
```

File: src/data/reference-data/reference-metadata.ts
Comment (nawatts) [inline src/data/reference-data/reference-metadata.ts]: Nit: can we call this `getReferenceLabel` to avoid introducing a new term: "alias". I think "label" is already well established for what we're trying to do here: separate user facing text from underlying IDs.

```suggestion
export const getReferenceLabel = (referenceName) => {
```

**Review by nawatts — CHANGES_REQUESTED** [review]

**Comment by rickymagner** [comment]
Thanks @nawatts again for those very helpful comments. I just pushed a commit implementing the suggested changes. In particular, using React's Select's ability to distinguish between `value` and `label` in its option was very useful.

There are now no references to aliases (hopefully) anywhere in favor of the value-label dichotomy. The code is still flexible enough to allow for future changes to the labeling paradigm for references.

**Review by nawatts — APPROVED** [review]
I think we'll still need approval from product, but this seems like a nice low-risk improvement to me.

**Review by davidangb — APPROVED** [review]
thank you for this! I do see conflicts that need to be resolved before merge, unfortunately.

**Comment by nawatts** [comment]
> I do see conflicts that need to be resolved before merge, unfortunately.

Oh, I didn't think about this PR when I renamed src/data to src/workspace-data in #4266. Hopefully should be easy to resolve.

**Comment by rickymagner** [comment]
Thanks @davidangb for the review! And yeah @nawatts the name change caused some merge conflicts but was very easy to fix. Once the tests finish it should be good to go. Thanks!

**Comment by rickymagner** [comment]
Uh oh, it looks like a test failed now. I tried clicking the CircleCI button above and it was something in the UI Integration tests, but I'm having trouble parsing the test output. Any ideas @nawatts or @davidangb ? It looks like a timeout issue?
```
    thrown: "Exceeded timeout of 120000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      30 | const registerTest = ({ fn, name, timeout = defaultTimeout, targetEnvironments = _.keys(envs) }) => {
      31 |   return _.includes(environment, targetEnvironments)
    > 32 |     ? test(name, () => withPageLogging(withScreenshot(name)(fn))({ context, page, ...targetEnvParams }), timeout)
         |       ^
      33 |     : test(name, () => console.log(`Skipping ${name} as it is not configured to run on the ${environment} environment`), timeout);
      34 | };
      35 |

      at registerTest (utils/jest-utils.js:32:7)
      at Object.<anonymous> (tests/export-workspace-dataset-to-workspace.js:14:1)
```
Not sure if this is transient or a genuine bug somewhere? It suggests a screenshot is saved somewhere but not sure how to access that.

**Comment by nawatts** [comment]
It's likely transient. I re-ran the integration tests.

Screenshots can be found on the CircleCI job's artifacts page
https://app.circleci.com/pipelines/github/DataBiosphere/terra-ui/18953/workflows/64ad204f-310c-4eb2-9c07-14f75b03eaad/jobs/68835/artifacts

**Comment by nawatts** [comment]
FYI, all changes need to have a JIRA ticket associated. I've filed an AJ ticket ([AJ-1352](https://broadworkbench.atlassian.net/browse/AJ-1352)) for this one.

[AJ-1352]: https://broadworkbench.atlassian.net/browse/AJ-1352?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by rickymagner** [comment]
Thanks! In the future I can make a ticket for my own PRs and then self-assign, if that seems right.

**Pull request merged**
